homeassistant/components/solaredge/sensor.py

```diff
@@ -426,20 +426,24 @@ def update(self):
         self.unit = energy_details["unit"]
         meters = energy_details["meters"]
 
-        for entity in meters:
-            for key, data in entity.items():
-                if key == "type" and data in [
-                    "Production",
-                    "SelfConsumption",
-                    "FeedIn",
-                    "Purchased",
-                    "Consumption",
-                ]:
-                    energy_type = data
-                if key == "values":
-                    for row in data:
-                        self.data[energy_type] = row["value"]
-                        self.attributes[energy_type] = {"date": row["date"]}
+        try:
+            for entity in meters:
+                for key, data in entity.items():
```

Comment:
This code is really weird. If you look at the code, it just does this:

```python
last_data = entity["values"][-1]
self.data[entity["type"]] = last_data["value"]
self.attributes[entity["type"]] = {"date": last_data["date"]}
```

This should be cleaned up first.

Reply:
Thank for the tip, but, you know, I'm not so skilled python dev. I've changed the code in accordance so what you've recommended.  But I need more time to understand what are you writing here, thank you.